### PR TITLE
improve examples

### PR DESCRIPTION
**Pull request by nicolas-grekas**

### CODE REVIEW

File: src/Command/HttpClientAsync.php [hunk 1]
```diff
@@ -35,45 +35,33 @@ protected function configure()
...
     {
-        $delays = [1, 3, 3, 5, 10];
+        // this command runs requests concurrently and read them in *network* order in 23s
+
+        $delays = [10, 3, 3, 5, 1];
         $responses = [];
 
         foreach ($delays as $offset => $delay) {
-            try {
-                $responses[] = $this->httpClient->request(
```
Comment (nicolas-grekas) [inline src/Command/HttpClientAsync.php:43]: the request method is lazy: it should do no network, no need to wrap it in a try/catch

File: src/Command/HttpClientAsync.php [hunk 2]
```diff
@@ -35,45 +35,33 @@ protected function configure()
...
-                $output->writeln(sprintf('<error>Streaming failed: %s</error>', $chunk->getError()));
-
-                continue;
-            }
+        foreach ($stream as $response => $chunk) {
+            try {
+                if ($chunk->isLast()) {
+                    $jsons[$response->getInfo('url')] = $response->toArray();
+                }
+            } catch (TransportExceptionInterface $transportException) {
```
Comment (nicolas-grekas) [inline src/Command/HttpClientAsync.php:56]: chunks do throw instead

File: src/Command/HttpClientBatch.php [hunk 3]
```diff
@@ -34,29 +34,29 @@ protected function configure()
...
+        }
+
+        /** @var ResponseInterface $response */
+        foreach ($responses as $url => $response) {
             try {
-                $responses[] = $this->httpClient->request(
-                    'GET',
-                    sprintf('https://reqres.in/api/users/%d?delay=%d', $offset + 1, $delay)
-                );
+                $data = $response->toArray();
```
Comment (nicolas-grekas) [inline src/Command/HttpClientBatch.php:50]: responses throw, not the request method (unless one is using a non-lazy implementation, but hey don't use any of them if one exists :) )

**Review by nicolas-grekas — COMMENTED** [review]
The added comments highlight a major difference: reading the responses can happen in *network* order when using the Symfony client. (and you can also have access to individual chunks of incoming data, but none of your examples are about that so it's a bit off-topic.)

**Review by dbrumann — COMMENTED** [review]
Thanks for the improvement and the explanations

**Pull request merged**